### ISSUE TITLE
📖 [Story] Porting Testing to GHA

### ISSUE THREAD

**Issue by narendasan** [pytorch/TensorRT#1946]
## TL;DR

We need to move our CI to GHA based CI.

## Goal(s)

Move our CI including testing and artifact building to GHA 

## Implementation Phases

1. Build automation
2. Testing automation
3. CD and release automation

## Additional context

- https://github.com/pytorch/test-infra/wiki/Using-Nova-Reusable-Build-Workflows
- https://github.com/pytorch/test-infra/wiki/Writing-generic-CI-jobs
- https://docs.github.com/en/actions/migrating-to-github-actions/automated-migrations/migrating-from-circleci-with-github-actions-importer
- https://docs.github.com/en/actions/migrating-to-github-actions/manual-migrations/migrating-from-circleci-to-github-actions
